src/screens/BlaireHome/BlaireHome.tsx

```diff
@@ -1,20 +1,130 @@
+  {matched: false, src: require('./assets/yellow.png')},
+  {matched: false, src: require('./assets/purple.png')},
+];
+
+type Card = {
+  id?: number;
+  src: ImageSourcePropType;
+  matched: boolean;
+};
+
 type Props = {};
 
 function BlaireHome(props: Props) {
-  const onPress = useCallback(() => {
+  const [cards, setCards] = useState<Card[]>([]);
+  const [turns, setTurns] = useState<number>(0);
+  const [pairs, setPairs] = useState<number>(NUM_PAIRS);
+  const [choiceOne, setChoiceOne] = useState<Card | null>(null);
+  const [choiceTwo, setChoiceTwo] = useState<Card | null>(null);
+
+  const onPressDinTaiFung = useCallback(() => {
     Linking.openURL('https://dintaifungusa.com/');
   }, []);
+
+  const shuffleCards = useCallback(() => {
+    const shuffledCards = [...cardImages, ...cardImages]
+      .sort(() => Math.random() - 0.5)
+      .map((card) => ({...card, id: Math.random()}));
+
+    setCards(shuffledCards);
+    setTurns(0);
+  }, []);
+
+  const resetChoices = useCallback(() => {
+    setChoiceOne(null);
+    setChoiceTwo(null);
+    setTurns((prevTurns) => prevTurns + 1);
+  }, []);
+
+  useEffect(() => {
+    if (choiceOne != null && choiceTwo != null) {
+      if (choiceOne.src === choiceTwo.src) {
+        setCards((prevCards) => {
+          return prevCards.map((card) => {
+            if (card.src === choiceOne.src) {
+              return {...card, matched: true};
+            }
+            return card;
+          });
+        });
+        setPairs((prevPairs) => prevPairs - 1);
+        if (pairs === 1) {
+          Alert.alert('Congrats', `You finished the game in ${turns} turns!`, [
+            {text: 'OK', onPress: () => shuffleCards()},
+          ]);
+          return;
+        }
+        resetChoices();
+      } else {
+        setTimeout(() => {
+          resetChoices();
+        }, 1000);
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [choiceOne, choiceTwo, resetChoices]);
+
+  const onPressCard = useCallback(
+    (card) => () => {
+      choiceOne == null ? setChoiceOne(card) : setChoiceTwo(card);
+    },
+    [choiceOne],
+  );
+
+  const renderItem: ListRenderItem<Card> = useCallback(
+    ({item, index}) => {
+      return (
+        <SingleCard
+          cardImageSource={item.src}
+          flipped={item === choiceOne || item === choiceTwo || item.matched}
+          key={item.id}
+          onPress={onPressCard(item)}
+        />
+      );
+    },
+    [choiceOne, choiceTwo, onPressCard],
+  );
+
+  const keyExtractor = useCallback((item: Card) => {
+    return String(item.id);
+  }, []);
+
   return (
     <View style={styles.container}>
       <Text style={styles.text}>{`B L A I R E`}</Text>
-      <Button onPress={onPress} title="Press me" />
+      <Button
+        onPress={onPressDinTaiFung}
+        style={{marginTop: 20}}
+        title="Press me"
+      />
+      <Button
+        onPress={shuffleCards}
+        style={{marginBottom: 20, marginTop: 20}}
```

Comment:
any static styles should be put inside the corresponding `styles.ts` file of a component

does the linter not warn you about static styles not being created as a `StyleSheet`? i think this should be a lint rule